### PR TITLE
Update removing-sensitive-data-from-a-repository.md

### PR DESCRIPTION
**Pull request by diberry**
Add gitleaks as another way to prevent secrets leaks.

### Why:

Closes: 

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@diberry Thanks so much for opening a PR! I'll get this triaged for review ✨

File: content/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository.md [hunk 1]
```diff
@@ -218,6 +218,7 @@ There are a few simple tricks to avoid committing things you don't want committe
 - Avoid the catch-all commands `git add .` and `git commit -a` on the command line—use `git add filename` and `git rm filename` to individually stage files, instead.
 - Use `git add --interactive` to individually review and stage changes within each file.
 - Use `git diff --cached` to review the changes that you have staged for commit. This is the exact diff that `git commit` will produce as long as you don't use the `-a` flag.
+- Use [gitleaks](https://github.com/gitleaks/gitleaks) to detect and prevent commits which contain hardcoded secrets.
```
Comment (jc-clark) [inline content/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository.md:221]:
```suggestion
- Enable push protections for your repository to detect and prevent commits which contain hardcoded secrets. For more information, see "[AUTOTITLE](/code-security/secret-scanning/push-protection-for-repositories-and-organizations#about-push-protection-for-repositories-and-organizations)."
```

**Review by jc-clark — DISMISSED** [review]
Thanks for the suggestion to add a preventative measure to this list @diberry! 

After checking in with our team, we agreed that this information is good to include here, but we can't document the third-party `gitleaks` tool. I suggested changes for enabling push protections for the repository instead. 

What do you think? If you commit the changes, I can help merge.

File: content/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository.md [hunk 2]
```diff
@@ -218,6 +218,7 @@ There are a few simple tricks to avoid committing things you don't want committe
 - Avoid the catch-all commands `git add .` and `git commit -a` on the command line—use `git add filename` and `git rm filename` to individually stage files, instead.
 - Use `git add --interactive` to individually review and stage changes within each file.
 - Use `git diff --cached` to review the changes that you have staged for commit. This is the exact diff that `git commit` will produce as long as you don't use the `-a` flag.
+- Enable push protections for your repository to detect and prevent pushes which contain hardcoded secrets. For more information, see "[AUTOTITLE](/code-security/secret-scanning/push-protection-for-repositories-and-organizations#about-push-protection-for-repositories-and-organizations)."
```
Comment (mchammer01) [inline content/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository.md:221]: Thanks @diberry and @jc-clark 
I work on the docs for push protection and have slightly reworded the suggestion.

The feature is called "push protection" and what's great with it, is that it prevents pushes with accidentally committed secrets from being leaked and from being committed onto the codebase ✨  

```suggestion
- Enable push protection for your repository to detect and prevent pushes which contain hardcoded secrets from being committed. For more information, see "[AUTOTITLE](/code-security/secret-scanning/push-protection-for-repositories-and-organizations#about-push-protection-for-repositories-and-organizations)."
```

File: content/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository.md [hunk 3]
```diff
@@ -216,6 +216,7 @@ There are a few simple tricks to avoid committing things you don't want committe
 - Avoid the catch-all commands `git add .` and `git commit -a` on the command line—use `git add filename` and `git rm filename` to individually stage files, instead.
 - Use `git add --interactive` to individually review and stage changes within each file.
 - Use `git diff --cached` to review the changes that you have staged for commit. This is the exact diff that `git commit` will produce as long as you don't use the `-a` flag.
+- Enable push protection for your repository to detect and prevent pushes which contain hardcoded secrets from being committed. For more information, see "[AUTOTITLE](/code-security/secret-scanning/push-protection-for-repositories-and-organizations#about-push-protection-for-repositories-and-organizations)."
```
Comment (mchammer01) [inline content/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository.md:219]: Sorry, forgot about this
```suggestion
- Enable push protection for your repository to detect and prevent pushes which contain hardcoded secrets from being committed to your codebase. For more information, see "[AUTOTITLE](/code-security/secret-scanning/push-protection-for-repositories-and-organizations#about-push-protection-for-repositories-and-organizations)."
```

**Review by mchammer01 — APPROVED** [review]
This LGTM :shipit:

**Pull request merged**